app/controllers/holds_controller.rb

```diff
@@ -122,9 +123,10 @@ def error_message(exception)
 
   def update
     LogWrapper.log('DEBUG', {'message' => 'update.start', 'method' => 'app/controllers/holds_controller.rb.update'})
-
     @hold = Hold.find_by_access_key(params[:id])
-
+    # Update teacher-set available copies while cancelling the hold.
```

Comment:
Is update only used for cancels? If not you don't want to change `available_copies` here, right?

Reply:
@thisisstephenbetts  Yes Stephen only update method for holds cancellation. I checked more times. I will add it in comments and yes i will add test cases also.

Reply:
@thisisstephenbetts Yes. I verified more times, when we cancel the hold in sets page, request is coming to update method only.

http://qa-www.mylibrarynyc.org/app#/holds/3861795ae745d91d7b08d3540/cancel

Reply:
@thisisstephenbetts  here update method used for cancellation only

Reply:
I'm only confused because underneath it say `if c{:status] == 'cancelled` implying that this method _could_ be used for something else. It might be more future proof if you moved this calcuation inside that block, so it's close (maybe after) the hold cancel (and we don't want to adjust the total if the cancel fails, so that's another reason).

Reply:
@thisisstephenbetts  Sorry.yes I moved now.